### PR TITLE
Added add litigation friend support for solicitor 2

### PR DESCRIPTION
**Pull request by Gareth40343**
https://tools.hmcts.net/jira/browse/CMC-1919

Added support for notifying respondent 2 solicitor when litigation friend.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java [hunk 1]
```diff
@@ -99,8 +111,15 @@ private CallbackResponse aboutToSubmitMultiparty(CallbackParams callbackParams)
         } else {
             CaseData.CaseDataBuilder caseDataUpdated = caseData.toBuilder()
                 .businessProcess(BusinessProcess.ready(ADD_DEFENDANT_LITIGATION_FRIEND));
-            String selectedOption = Optional.of(caseData).map(CaseData::getSelectLitigationFriend)
+
+            boolean diffSolicitors = caseData.getRespondent2SameLegalRepresentative() == null
+                || caseData.getRespondent2SameLegalRepresentative() == NO;
```
Comment (sherlynkhaw) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java:116]: Do you think it's possible to use flow flags from stateflowengine?

FlowFlag.TWO_RESPONDENT_REPRESENTATIVES?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java [hunk 2]
```diff
@@ -132,6 +151,7 @@ private CallbackResponse aboutToSubmitMultiparty(CallbackParams callbackParams)
                 .data(caseDataUpdated.build().toMap(objectMapper))
                 .build();
         }
+
```
Comment (sherlynkhaw) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java:154]: eww

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandlerTest.java [hunk 3]
```diff
@@ -81,6 +99,67 @@ void shouldPrepopulateDynamicListWithOptions_whenInvoked() {
 
             assertTrue(response.getData().containsKey("selectLitigationFriend"));
         }
+
+        @Test
+        void shouldNotSetSIsRespondent1_whenInvoked() {
```
Comment (sherlynkhaw) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandlerTest.java:104]: shouldNotSetS?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java (hunk 1 above)
Comment (Gareth40343) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java:116]: That's a good idea. Will give it ago.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java (hunk 2 above)
Comment (Gareth40343) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandler.java:154]: Haha agreed. Will remove this.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandlerTest.java (hunk 3 above)
Comment (Gareth40343) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/AddDefendantLitigationFriendCallbackHandlerTest.java:104]: Will fix this.

**Review by bhssamray — DISMISSED** [review]

**Review by HarryH96 — APPROVED** [review]

**Review by sherlynkhaw — APPROVED** [review]

**Review by sherlynkhaw — COMMENTED** [review]
in eventhistorymapper.java on line 154:
you may need to add respondent2litigationfriendevent or something?

**Review by sherlynkhaw — DISMISSED** [review]

**Review by sherlynkhaw — APPROVED** [review]

**Review by bhssamray — APPROVED** [review]

**Pull request merged**